### PR TITLE
check simple vote could have 1 or 2 signatures when creating sanitized transaction

### PR DESCRIPTION
**Pull request by taozhu-chicago**
#### Problem
When replay stage flags simple vote transaction, it should check number of signatures as banking stage does [here](https://github.com/solana-labs/solana/blob/master/perf/src/sigverify.rs#L355-L359). 

#### Summary of Changes
- add tests
- add signature count check

Fixes #

### CODE REVIEW

File: perf/src/sigverify.rs [hunk 1]
```diff
@@ -1410,6 +1410,21 @@ mod tests {
             check_for_simple_vote_transaction(&mut packet, &packet_offsets, 0).ok();
             assert!(!packet.meta().is_simple_vote_tx());
         }
+
+        // single legacy vote tx with extra (invalaid) signature is not
```
Comment (apfitzge) [inline perf/src/sigverify.rs:1414]: typo:
```suggestion
        // single legacy vote tx with extra (invalid) signature is not
```

File: sdk/src/transaction/sanitized.rs [hunk 2]
```diff
@@ -294,3 +297,57 @@ impl SanitizedTransaction {
...
+#[allow(clippy::integer_arithmetic)]
+mod tests {
+    use {
+        super::*,
+        crate::signer::{keypair::Keypair, Signer},
+        solana_program::vote::{self, state::Vote},
+    };
+
+    #[test]
+    fn test_try_create_simple_vote_tx() {
```
Comment (apfitzge) [inline sdk/src/transaction/sanitized.rs:311]: `is_simple_vote_tx` is an optional argument to the `try_create` fn.

Logically, these shouldn't ever be different. But the behavior is currently the passed value takes precedence.

Should the test(s) make sure that is the case?

**Review by apfitzge — CHANGES_REQUESTED** [review]
Thanks for the fix, gets these 2 checks to be consistent.

1 typo suggestion & a question about tests.

More high-level, I'm trying to think of a better way, since this still is open to inconsistency if one of the 2 impls changes. They do not directly share a data type, one is packet vs sanitized message, so this is tricky.

**Comment by taozhu-chicago** [comment]
> More high-level, I'm trying to think of a better way, since this still is open to inconsistency if one of the 2 impls changes. They do not directly share a data type, one is packet vs sanitized message, so this is tricky.

was thinking if `try_create()` can take `bool` instead of `Option<bool>` when creating from unsanitized versioned transaction. Looks like only few places (rpc.rs and bank.rs) actually used `None`. Maybe it is possible, leaving it to another time.

**Review by apfitzge — APPROVED** [review]
lgtm

**Pull request merged**